crates/swc_ecma_transforms_compat/src/es2015/computed_props.rs

```diff
@@ -308,6 +309,28 @@ impl VisitMut for ObjectLitFolder {
     }
 }
 
+fn used_define_enum_props<T: VisitWith<DefineEnumVisitor>>(node: &T) -> bool {
+    let mut visitor = DefineEnumVisitor::default();
+    node.visit_children_with(&mut visitor);
+    visitor.used_define_enum_props
+}
+
+#[derive(Default)]
+struct DefineEnumVisitor {
+    used_define_enum_props: bool,
+}
+
+impl Visit for DefineEnumVisitor {
```

Comment:
I think this visitor should be noop for nested functions.

```ts
const obj = {
    foo() {
        const obj2 = {
            get [1]() {}
        }
    }
};
```

The inputs like this can be broken.

Reply:
Yes, this is a point worth considering.
This visitor should not visit the nest functions.

There is another problem.
ComputedProps will recursively access the inner AST, and collect all inner VarDeclarator.
So VarDeclarator is misplaced to the outermost layer.

```diff
-   var _obj, _mutatorMap = {};
    var obj1 = {
        foo: function() {
+   var _obj, _mutatorMap = {};
            var obj2 = (
                _obj = {}, 
                _mutatorMap[1] = _mutatorMap[1] || {}, 
                _mutatorMap[1].get = function() {}, 
                _defineEnumerableProperties(_obj, _mutatorMap), 
                _obj
            );
        }
    };
```

I merged the ObjectLitFolder into ComputedProps to ensure that the VarDeclarator is also collected and updated each time the internal stmts are accessed.